packages/docs-site/blog/2023/06/29/new-language-features.md

```diff
@@ -0,0 +1,302 @@
+<script setup>
+import BlogMeta from "../../../../src/components/BlogMeta.vue";
+</script>
+
+# What Have We Done to the Languages?
+
+<BlogMeta github="liangyiliang" date="2023-06-13" />
+
+I joined the Penrose team in July 2022, and worked on of a few language-related improvements to Penrose which have the potential of making Penrose more flexible.
+
+## Binary Symmetric Predicates
+
+In July 2022, we implemented support for binary symmetric predicates which allowed for more flexible style-substance matching. By marking a binary predicate with the `symmetric` keyword, the Domain author basically tells Penrose that it does not care about the order of input to the predicate, just like a symmetric mathematical relation.
+
+### Why did we need this?
+
+The necessity of this feature was pointed out in [Professor Keenan Crane](https://www.cs.cmu.edu/~kmcrane/)'s [GitHub issue](https://github.com/penrose/penrose/issues/744) using an interesting example. Consider the Domain program
+
+```
```

Comment:
Add syntax highlighting to this and the rest of your code blocks.

Reply:
+1. Also can the honorary title because this is meant to be pretty casual

Reply:
Done